File: Menza-WA/src/main/java/hr/tvz/zr/menzawa/model/Blagajna.java
package hr.tvz.zr.menzawa.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Entity
@Data
@Table(name = "BLAGAJNE")
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Blagajna{

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "korisnicko_ime")
    private String username;
    @Column(name = "lozinka")
    private String password;
    private String naziv;
    @JsonBackReference
    @ManyToOne
    @JoinColumn(name= "menza_id")
    private Menza menza;

    public UUID getId() {
        return id;
    }

}

File: Androidmenza/app/src/main/java/hr/tvz/android_menza/Blagajna.kt
package hr.tvz.android_menza

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize
import java.util.UUID

@Parcelize
data class Blagajna(
    @SerializedName("id") var id: UUID,
    @SerializedName("username") var username: String,
    @SerializedName("password") var password: String
): Parcelable{

}
